restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -151,6 +151,13 @@ public boolean maybeCreateTable(String keyspaceName, String tableName) {
       throw new ErrorCodeRuntimeException(ErrorCode.DATASTORE_KEYSPACE_DOES_NOT_EXIST, message);
     }
 
+    // if table exists, do nothing
+    Table table = ks.table(tableName);
+    if (table != null) {
+      return false;
+    }
+
+    // otherwise, check that the name is  valid
     if (!tableName.matches("^[a-zA-Z0-9_]+$")) {
```

Comment:
Should this check be left to the datastore to validate? 

```
token@cqlsh> CREATE TABLE ks."non-standard-name?" (k int PRIMARY KEY, v int);
ConfigurationException: Table name must not be empty, more than 222 characters long, or contain non-alphanumeric-underscore characters (got "non-standard-name?")
```

We can catch that `ConfigurationException` error and change it to what you have here. This check is also missing the 222 character length limitation so maybe it makes sense to let the data store validate since it's more restrictive?

Reply:
I am up for this, @EricBorczuk do you see any problem with that?

Reply:
What are we doing about this one @EricBorczuk ? I am also fine to leave it as it is and spare the request to the data store if we know that the name is invalid?

Reply:
OK I am merging as it is, saving the data store request if we already know that the table name is not valid.